editor.planx.uk/src/@planx/components/Checklist/model.ts

```diff
@@ -123,29 +123,38 @@ export const getLayout = ({
   return ChecklistLayout.Basic;
 };
 
-export const checklistValidationSchema = ({
-  allRequired,
-  options,
-  groupedOptions,
-}: Checklist) => {
+export const checklistInputValidationSchema = ({
+  data: { allRequired, options, groupedOptions },
+  required,
+}: {
+  // Cannot use FieldValidationSchema<ChecklistInput> as this is a simplified representation (i.e. no groups)
+  data: Checklist;
+  required: boolean;
+}) => {
   const flatOptions = getFlatOptions({ options, groupedOptions });
 
   return array()
-    .required()
+    .when([], {
+      is: () => required,
+      then: array().required(),
+      otherwise: array().notRequired(),
+    })
     .test({
       name: "atLeastOneChecked",
       message: "Select at least one option",
       test: (checked?: Array<string>) => {
+        if (!checked?.length) return true;
+
         return Boolean(checked && checked.length > 0);
       },
     })
     .test({
       name: "notAllChecked",
       message: "All options must be checked",
       test: (checked?: Array<string>) => {
-        if (!allRequired) {
-          return true;
-        }
+        if (!checked?.length) return true;
+        if (!allRequired) return true;
```

Comment:
Despite the similar names I believe the interaction between these two properties is correct -

- If there are no checked items - don't run this test, defer to the required/not required `when()` statement above
- If `allRequired` is set, all values must be checked, even if the field is optional

This means there's two possible states for an optional checklist with `allRequired` - all empty, or all checked.

I think this permutation is unlikely to be used, but I'd like a sense check here that I'm not introducing some odd/unexpected behaviour.